### ISSUE TITLE
Wrong AppVeyor badge

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/crates.io#693]
https://crates.io/crates/serde is showing a badge from:

    https://ci.appveyor.com/api/projects/status/github/serde-rs/serde?svg=true&branch=master

The link is correct and shows that the build is passing, but the badge says "unknown".

AppVeyor tells me to use the following badge:

    https://ci.appveyor.com/api/projects/status/avsuce1irt8h63u2/branch/master?svg=true

Unclear how this is generated but it would be nice if I could have crates.io show the correct badge.

Implementation steps added by @carols10cents:

- [ ] Add an `id` field with type `Option<String>` to [the Appveyor badge enum variant](https://github.com/rust-lang/crates.io/blob/405b2ad74715f22d756de749be4acb9626c1019b/src/badge.rs#L16-L20)
- [ ] Add `id: alias('badge.attributes.id')` to the [appveyor badge component](https://github.com/rust-lang/crates.io/blob/405b2ad74715f22d756de749be4acb9626c1019b/app/components/badge-appveyor.js)
- [ ] Add an `image-url` attribute, computed from the id to the appveyor badge component that checks to see if `id` is defined; if it is, return `https://ci.appveyor.com/api/projects/status/{{ id }}/branch/{{ branch }}?svg=true`. If not, return `https://ci.appveyor.com/api/projects/status/{{ service }}/{{ repository }}?svg=true&branch={{ branch }}` (that's what [the appveyor badge template currently uses for the image src](https://github.com/rust-lang/crates.io/blob/405b2ad74715f22d756de749be4acb9626c1019b/app/templates/components/badge-appveyor.hbs#L3))
  - I've used handlebars interpolation here but you should use js interpolation, I'm being lazy sorry sorry!
- [ ] Change [the appveyor badge template](https://github.com/rust-lang/crates.io/blob/405b2ad74715f22d756de749be4acb9626c1019b/app/templates/components/badge-appveyor.hbs#L3) to use the `image-url` attribute for the img src instead
- [ ] Publish a crate to your local instance that has the URLs dtolnay has listed for serde and manually verify the badge is displayed correctly (we used to have frontend tests for badge tests but they got deleted?)
- [ ] Send a pull request to cargo to [update the badge documentation](https://github.com/rust-lang/cargo/blame/master/src/doc/manifest.md#L187-L190) to indicate you can specify the appveyor project id if you want to use that instead

**Comment by carols10cents** [comment]
Hm, these docs: https://github.com/appveyor/website/blob/master/src/docs/status-badges.md#badges-for-projects-with-public-repositories-on-github-and-bitbucket say that we're allowed to use that URL :-/

![Serde status](https://ci.appveyor.com/api/projects/status/github/serde-rs/serde?svg=true&branch=master)

**Comment by dtolnay** [comment]
> You can't prevent someone from creating a duplicate project.

Feature request: given that the inferred badge URL is unreliable, I would like to be able to set the project id and get a reliable badge.

```toml
[badges]
appveyor = { repository = "serde-rs/serde", id = "avsuce1irt8h63u2" }
```

**Comment by carols10cents** [comment]
:( i suppose we'll have to do that :(

**Comment by mattgathu** [comment]
Hi @dtolnay @carols10cents I have made a PR that addresses this issue. 🙂

**Comment by SeanPrashad** [comment]
@dtolnay @carols10cents Looks like this was supposed to be closed with #1050!
